trafficFlow/carFollowingModel/roads/circularRoad.py

```diff
@@ -18,16 +18,24 @@ class CircularRoad(BaseRoad):
     def __init__(self):
         super().__init__()
 
-    def get_distance(self, position1, position2, lane1, lane2):
-        if lane1 == lane2:
-            if position1 < position2:
-                return position1 - position2 + lane1.full_length
-            return position1 - position2
-        else:
-            ###########################################################
-            raise NotImplementedError
-
     def get_position(self, position, lane):
         if position < lane.full_length:
             return position
         return position - lane.full_length
+
+    def get_distance(self, vehicle1, vehicle2):
+        if not vehicle1:
+            return vehicle2.lane.full_length  # math.inf
+        if not vehicle2:
+            return vehicle1.lane.full_length  # math.inf
+        if vehicle1.position >= vehicle2.position:
+            return vehicle1.position - vehicle2.position
+        else:
+            return vehicle1.position - vehicle2.position + vehicle1.lane.full_length
+
+    def between(self, first, second, third):
```

Comment:
This could use some documentation, which one should be between the two others. I guess you want to check whether third is between first and second. In that case, what does the first condition do?

Reply:
Actually, I want to check whether the second is between first and third. In detail, first is assumed to be the first vehicle, second the second one and third the last. The problem or difficulty here is that we have a circular road which makes everything a bit different. The positions are always between 0 and `lane.full_length` as the position gets shifted back when it exceeds `lane.full_length`. Therefore there occur different cases when we want to determine the "between"-relation. Maybe I'm still missing something here, but as far as I can see, this should be correct.

To make it a bit more clear, consider the following cases:
first=300, second=200, third=100 -> True (first condition, first case of "or")
first=300, second=200, third=900 -> True (first condition, second case of "or")
first=100, second=900, third=500 -> True (second condition)
first=300, second=400, third=900 -> False

Reply:
Okay, makes sense. But you should still add that documentation as a comment to the function.

Reply:
That function is unfortunately not really self-explaining. I will document it while documenting all the other changes. This should be done before merging into master of course.